File: 1021.cpp
class Solution {
public:
    string removeOuterParentheses(string S) {
        int tmp = 0;
        int begin = 0;
        string res = "";
        for(int i = 0; i < S.length(); i++){
            if(S[i] == '(')
                tmp++;
            else
                tmp--;
            if(tmp == 0){
                res += S.substr(begin + 1, i  - begin - 1);
                begin = i + 1;
            }
        }
        return res;
    }
};

File: 1021.py
class Solution:
    def removeOuterParentheses(self, S: str) -> str:
        res = ""
        tmp = 0
        flag = False
        for ch in S:
            if '(' == ch:
                tmp += 1
            else:
                tmp -= 1
            
            if 0 != tmp:
                if True == flag:
                    res += ch
                else:
                    flag = True
            else:
                flag = False
        return res
